app/src/ui/history/compare.tsx

```diff
@@ -413,41 +415,51 @@ export class CompareSidebar extends React.Component<
     this.setState({ filterText: text })
   }
 
-  private onSelectionChanged = (branch: Branch | null) => {
+  private clearFilterState = () => {
+    this.setState({
+      selectedBranch: null,
+      filterText: '',
+      compareType: CompareType.None,
+    })
+
+    this.props.dispatcher.loadCompareState(
+      this.props.repository,
+      DisplayHistory
+    )
+  }
+
+  private onSelectionChanged = (
+    branch: Branch | null,
+    source: SelectionSource
```

Comment:
@iAmWillShepherd just a heads up, I think we've gone back and forth about this in the past but I think that the lack of `source` here meant the selection events from `FilterList` were a bit confusing. Hope that helps with consuming the component now as part of the Compare list.

Reply:
Oh this is great!!! You actually solved the thing I was working through today. I didn't think about adding a source param to that function. I thought I had did something else wrong and decided to look at how I passed the textbox component to the filter-list component.